fe/src/main/java/org/apache/doris/analysis/CaseExpr.java

```diff
@@ -251,4 +257,72 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
         }
         return exprs;
     }
+
+    // this method just compare literal value and not completely consistent with be,for two cases
+    // 1 not deal float
+    // 2 just compare literal value with same type. for a example sql 'select case when 123 then '1' else '2' end as col'
+    //      for be will return '1', because be only regard 0 as false
+    //      but for current LiteralExpr.compareLiteral, `123`' won't be regard as true
+    //  the case which two values has different type left to be
+    public static Expr computeCaseExpr(CaseExpr expr) {
+        LiteralExpr caseExpr;
+        int startIndex = 0;
```

Comment:
would better rename `startIndex ` and `size `.

The `size` compute logic seems not consistent with BE ？

```
        int loop_start = has_case_expr() ? 1 : 0; \
        int loop_end = (has_else_expr()) ? num_children - 1 : num_children; \
```

Reply:
yes, be logic is right.I'll fix it